monai/transforms/utility/dictionary.py

```diff
@@ -327,6 +327,44 @@ def __call__(self, data):
         return d
 
 
+class CopyKeysd(MapTransform):
+    """
+    Copy specified keys from data dictionary and save with different key names.
+    It can copy several keys together and copy several times.
+
+    """
+
+    def __init__(self, keys: KeysCollection, times: int, names):
+        """
+        Args:
+            keys: keys of the corresponding items to be transformed.
+                See also: :py:class:`monai.transforms.compose.MapTransform`
+            times: expected copy times, for example, if keys is "img", times is 3,
+                it will add 3 copies of "img" data to the dictionary.
+            names(str, list or tuple of str): the names coresponding to the newly copied data,
+                the length should match `len(keys) x times`. for example, if keys is ["img", "seg"]
+                and times is 2, names can be: ["img_1", "seg_1", "img_2", "seg_2"].
+        """
+        super().__init__(keys)
+        if times < 1:
+            raise ValueError("times must be greater than 0.")
+        self.times = times
+        names = ensure_tuple(names)
+        if len(names) != (len(self.keys) * times):
+            raise ValueError("length of names does not match `len(keys) x times`.")
+        self.names = names
+
+    def __call__(self, data):
+        d = dict(data)
+        for idx, key in enumerate(self.keys):
```

Comment:
perhaps it's more readable with `self.keys * self.times`
```
for key, new_key in zip(self.keys * self.times, names):
     if new_key in d:
          raise KeyError
     d[new_key] = deepcopy(d[key])
```

Reply:
updated in the latest patch.
Thanks.